### DISCUSSION TITLE
null context error while trying to produce message with kafkaTemplate

### DISCUSSION THREAD

**Discussion by PatrickChauveau** [open-telemetry/opentelemetry-java-instrumentation#11728] [Q&A] [answered]
When I activate OpenTelemetry traces on a spring boot service that uses kafka i got this log output whenever the method KafkaTemplate.send() is called (I tried with & without the TracingProducerInterceptor but i get the same result):
Cannot invoke "io.opentelemetry.context.Context.makeCurrent()" because "context" is null

I am using fololowing components:
spring boot 3.1.8
spring-framework 6.0.16
spring-kafka 3.0.13
kafka client 3.4.1
opentelemetry-spring-boot-starter 2.5.0-alpha
opentracing-kafka-client 0.1.15

Please see logs & stacktrace attached.
[null_context.txt](https://github.com/user-attachments/files/16078548/null_context.txt)

**Comment by laurit** [comment]
@PatrickChauveau would it be possible for you to provide a minimal application that reproduces this issue. I fail to see what circumstances could lead to this `NullPointerException`.

**Comment by 5cr4b** [comment]
Hi @PatrickChauveau , Do you have any ideas about that? I'm encountering that and don't know how to resolve it?

**Reply by PatrickChauveau** [reply]
Hi, for now i have disabled this specific kafka traces: https://opentelemetry.io/docs/zero-code/java/agent/disable/#suppressing-specific-agent-instrumentation

I have not been able to get the time to build a minimal reproduction case yet...

**Comment by PatrickChauveau** [comment, marked as answer]
This error was a side effect of a double load of otel libraries. 
We had otel load at the start of the service with -javaagent option and we also had the import of `opentelemetry-spring-boot-starter` in our pom.xml. Removing the javaagent option solved this issue.

**Comment by Alice52** [comment]
f I must use agent and starter, how can I solve this problem?
